content/posts/2025-01-30-ebpf-flows-filtering.md

```diff
@@ -0,0 +1,229 @@
+|             | Range of ports to match on the flow or                   | for example 80-100                                                      |           |
+|             | Two ports to match on                                    | for example 80,100                                                      |           | 
+| icmpType    | ICMP type to match on the flow                           | for example 8 or 13                                                     |           |
+| icmpCode    | ICMP code to match on the flow                           | for example 0 or 1                                                      |           |
+| peerIP      | Peer IP to match on the flow                             | for example 1.1.1.1 or 1::1                                             |           |
+| peerCIDR    | Peer IP CIDR to match on the flow                        | for example 1.1.1.1/24 or 1::1/48                                       |           |
+| pktDrops    | filter flows with packets drop                           | true, false                                                             |           |
+| sampling    | sampling rate to use for filtered flows                  | for example 10 or 20 (any value >= 1)                                   |           |
+| tcpFlags    | TCP flags to filter flows by                             | "SYN";"SYN-ACK";"ACK";"FIN";"RST";"URG";"ECE";"CWR";"FIN-ACK";"RST-ACK" |           |
+
+
+Note:
+
+- A rule cannot include both ports and either sourcePorts or destPorts simultaneously.
+- The number of eBPF flow filter rules is capped at `16` to maintain efficient memory usage when this feature is enabled.
+- Validation webhook rejects rules containing duplicate CIDRs.
+- Supports both IPv4 and IPv6 formats.
+- If users wish to match any CIDR, the default `0.0.0.0/0` (Null CIDR) can be used.
+
+The example below demonstrates various filtering options using multi-rule filters.
+
+```yaml
+agent:
+  type: eBPF
+  ebpf:
+    flowFilter:
+      enable: true
+      rules:
+        - action: Accept
+          cidr: 10.128.0.0/24
+          peerCIDR: 10.129.0.0/24
+          ports: '443,6443'
+          protocol: TCP
+          sampling: 10
+        - action: Accept
+          cidr: 10.129.0.1/24
+          ports: 53
+          protocol: UDP
+          sampling: 20
+        - action: Reject
+          tcpFlags: SYN
+          cidr: 10.130.0.0/24
+          protocol: TCP
+          sourcePorts: 80-100
+        - action: Accept
+          cidr: 172.30.0.0/16
+          protocol: SCTP
+          pktDrops: true
+        - action: Reject
+          cidr: 8.8.8.8/32
+          protocol: ICMP
+          icmpType: 8 // ICMP Echo request packet
+```
+2️⃣ Packet Inspection – Extract relevant packet attributes within an eBPF program.
+
+3️⃣ Early Flow Filtering –
+Skip or Allow packets based on predefined conditions before doing further ebpf packets processing.
+The Following Diagram shows how eBPF filtering is done
```

Comment:
For the numbered list, I would suggest not using title case. Instead, use regular sentence case like this:
1. Defining filtering rules - 
2. Packet inspection - Extract relevant...
3. Early flow filtering - Skip or allow packets